agent/engine/task_manager.go

```diff
@@ -1141,7 +1142,23 @@ func (mtask *managedTask) startContainerTransitions(transitionFunc containerTran
 func (mtask *managedTask) startResourceTransitions(transitionFunc resourceTransitionFunc) (bool, map[string]string) {
 	anyCanTransition := false
 	transitions := make(map[string]string)
-	for _, res := range mtask.GetResources() {
+
+	resources := mtask.GetResources()
+	anyCanTransition, transitions = mtask.resourceTransitions(resources, transitionFunc, "",
+		anyCanTransition, transitions)
+
+	for _, cont := range mtask.Containers {
+		containerResources := cont.GetResources()
+		anyCanTransition, transitions = mtask.resourceTransitions(containerResources,
+			transitionFunc, "-"+cont.Name, anyCanTransition, transitions)
+	}
+
+	return anyCanTransition, transitions
+}
+
+func (mtask *managedTask) resourceTransitions(resources []taskresource.TaskResource, transitionFunc resourceTransitionFunc,
```

Comment:
Why does this function need to return the transitions map?

Reply:
This is to store all the transitions in the same transitions map, maybe using a recursive function is more readable??

Reply:
you can just pass the map created in 1144, it's the same map

Reply:
Updated, thanks!